plugins/modules/aci_l3out_extepg_to_contract.py

```diff
@@ -0,0 +1,338 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_l3out_extepg_to_contract
+short_description: Bind Contracts to External End Point Groups (EPGs)
+description:
+- Bind Contracts to External End Point Groups (EPGs) on ACI fabrics.
+options:
+  tenant:
+    description:
+    - Name of existing tenant.
+    type: str
+  l3out:
+    description:
+    - Name of the l3out.
+    type: str
+    aliases: ['l3out_name']
+  extepg:
+    description:
+    - Name of the external end point group.
+    type: str
+  contract:
+    description:
+    - Name of the contract.
+    type: str
+  contract_type:
+    description:
+    - The type of contract.
+    type: str
+    choices: ['consumer', 'provider']
+  priority:
+    description:
+    - This has four levels of priority.
+    type: str
+    choices: ['level1', 'level2', 'level3', 'unspecified']
+  provider_match:
+    description:
+    - This is configurable for provided contracts.
+    type: str
+    choices: ['all', 'at_least_one', 'at_most_one', 'none']
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment:
+- cisco.aci.aci
+
+notes:
+- The C(tenant), C(l3out) and C(extepg) must exist before using this module in your playbook.
+  The M(aci_tenant), M(aci_l3out) and M(aci_l3out_extepg) modules can be used for this.
+seealso:
+- name: APIC Management Information Model reference
+  description: More information about the internal APIC class B(tenant), B(l3extInstP) and B(l3extOut).
+  link: https://developer.cisco.com/docs/apic-mim-ref/
+author:
+- kudtarkar1 (@kudtarkar1)
+'''
+
+EXAMPLES = r'''
+- name: Bind External End Point Groups to Contracts
+  cisco.aci.aci_l3out_epg_to_contract:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    tenant: Auto-Demo
+    l3out: l3out
+    extepg : testEpg
+    contract: contract1
+    contract_type: provider
+    state: present
+  delegate_to: localhost
+
+- name: Remove existing contract to External End Point Groups
+  cisco.aco.aci_l3out_epg_to_contract:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    tenant: Auto-Demo
+    l3out: l3out
+    extepg : testEpg
+    contract: contract1
+    contract_type: provider
+    state: absent
+  delegate_to: localhost
+
+- name: Query the External End Point Groups
+  cisco.aci.aci_l3out_epg_to_contract:
+    host: apic
+    username: admin
+    password: SomeSecretePassword
+    l3out: l3out
+    state: query
+  delegate_to: localhost
+  register: query_result
+
+- name: Query all the External End Point Groups
```

Comment:
What do you return if I query for this? This seems to miss required parameters anyway (like contract_type). Do you have those in your test file?

Reply:
The test file has those parameters. Just need to change this example.

Reply:
I have not included this in the test file.